Upstream: yes
From 8fe81055762d9c9e6f03fb7853a985c94ef73ac3 Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Fri, 2 Sep 2016 14:45:26 +0900
Subject: [PATCH] scd: Release the card reader after card removal.

* scd/command.c (update_reader_status_file): Call apdu_close_reader.

--

GnuPG-bug-id: 2651
Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
---
 scd/command.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/scd/command.c b/scd/command.c
index 239480b..2909330 100644
--- a/scd/command.c
+++ b/scd/command.c
@@ -2340,7 +2340,10 @@ update_reader_status_file (int set_card_removed_flag)
 
           /* Set the card removed flag for all current sessions.  */
           if (vr->any && vr->status == 0 && set_card_removed_flag)
-            update_card_removed (idx, 1);
+	    {
+              apdu_close_reader (vr->slot);
+              update_card_removed (idx, 1);
+	    }
 
           vr->any = 1;
 
-- 
2.8.0.rc3

